### PR TITLE
[RSLC-2 ] Enable user to create targets

### PR DESCRIPTION
**Pull request by sntgrdr**
#### Board:
* [Ticket #2](https://trello.com/c/yLaBM2BZ/37-rslc-2-as-a-user-i-should-be-able-to-create-a-new-target-so-i-can-find-people-with-similar-interests-in-a-specific-area)
---
#### Description:
Created `target` model and controller and `create` and `show` endpoint into it
---
#### Notes:
*
---
#### Tasks:
- [x] Created target model
- [x] Created target controller and create and show endpoint
- [x] Created target policy
- [x] Created related tests
---
#### Risk:
*
---

### CODE REVIEW

**Review by sebastiancaraballo — COMMENTED** [review]
Nice work. Since this branch is on top of the other, I recommend changing this PR's base branch from `master` to `feature/create-topic-model-and-index-endpoint`.

File: app/controllers/api/v1/topics_controller.rb [hunk 1]
```diff
@@ -1,15 +1,9 @@
 module Api
   module V1
     class TopicsController < Api::V1::ApiController
-      before_action :topics
+      # uniqueness en name
```
Comment (brunoparma88) [inline app/controllers/api/v1/topics_controller.rb]: avoid comments

File: app/models/target.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#
+#  index_targets_on_topic_id  (topic_id)
+#  index_targets_on_user_id   (user_id)
+#
+class Target < ApplicationRecord
+  belongs_to :topic
+  belongs_to :user
+
+  validates :radius, presence: true,
+                     numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 1000 }
```
Comment (brunoparma88) [inline app/models/target.rb:25]: use constants

File: spec/requests/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    it 'creates a new target' do
+      expect {
+        subject
+      }.to change(Target, :count).by(1)
+    end
+
+    it 'returns the targets attributes' do
+      subject
+      expect(json['id']).to eq(Target.last.id)
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/targets/create_spec.rb:23]:
```suggestion
      expect(json[:id]).to eq(Target.last.id)
```

**Review by sebastiancaraballo — APPROVED** [review]
LGTM

**Review by brunoparma88 — APPROVED** [review]

**Pull request merged**